Src/Support/Google.Apis.Auth/OAuth2/UserCredential.cs

```diff
@@ -140,6 +140,10 @@ public async Task<AccessTokenWithHeaders> GetAccessTokenWithHeadersForRequestAsy
         }
         #endregion
 
+        /// <inheritdoc/>
+        public Task<OIdCToken> GetOIdCTokenAsync(OIdCTokenOptions options, CancellationToken cancellationToken = default) =>
+            Task.FromResult(new OIdCToken(_refreshManager));
+
```

Comment:
I do not think supporting OidcCredentials on UserCredentials is a good idea, unless we explicitly deny the default gcloud credentials.

see https://github.com/googleapis/google-auth-library-nodejs/issues/876

Reply:
OK, this is what I understood, but happy to remove it :)